File: simple/csvulnweb/CodeExec/SubClassTest.cs
using csvulnweb.Common;

namespace csvulnweb.CodeExec
{
    /// <summary>
    /// 代码执行漏洞 - 纯子类继承调用测试
    /// 调用链：请求参数 → VulnChild → VulnParent.VulnerableCodeExec → 漏洞触发点
    /// </summary>
    public class CodeExecSubClassTest : VulnChild
    {
        /// <summary>
        /// 子类新增的代码执行方法，内部调用父类漏洞方法
        /// 调用链：外部参数 → TestCodeExec → CallCodeExec(VulnChild) → VulnerableCodeExec(VulnParent)
        /// </summary>
        public string TestCodeExec(string userInput)
        {
            return CallCodeExec(userInput);
        }

        /// <summary>
        /// 子类硬编码漏洞测试方法
        /// 调用链：TestHardcodedCodeExec → CallHardcodedCodeExec(VulnChild) → HardcodedCodeExec(VulnParent)
        /// </summary>
        public string TestHardcodedCodeExec()
        {
            return CallHardcodedCodeExec();
        }
    }
}

File: simple/pythonvulnweb/code_exec/sub_class_test.py
from common.vuln_child import VulnChild

class CodeExecSubClassTest(VulnChild):
    def execute(self, code):
        # 代码执行漏洞测试：子类继承VulnChild，调用父类code_exec方法
        # 调用链：外部参数code -> CodeExecSubClassTest -> VulnChild -> VulnParent.code_exec -> eval
        # 漏洞点：eval执行用户控制的code参数
        return self.code_exec(code)
    
    def hardcoded_test(self):
        # 代码执行漏洞测试：硬编码场景，无外部参数
        # 调用链：hardcoded_test -> VulnChild -> VulnParent.hardcoded_code_exec -> eval
        return self.hardcoded_code_exec()